osal/efr32_wisun/osal_common.c

```diff
@@ -564,18 +564,28 @@ void osal_trickle_timer_start(timerid_t timerid, uint32_t imin, uint32_t imax, t
   }
 
   osal_gettimeofday(&tv, NULL);
-
-  seed = (((uint16_t)g_csmplib_eui64[6] << 8) | g_csmplib_eui64[7]);
-  srand(seed);
-  timers[timerid].t0 = tv.tv_sec + (rand()%imin);
+  
   timers[timerid].icur = imin;
   timers[timerid].imin = imin;
   timers[timerid].imax = imax;
   timers[timerid].is_running = true;
   timer_fired[timerid] = trickle_timer_fired;
-  min = timers[timerid].icur >> 1;
-  timers[timerid].tfire = timers[timerid].t0 + min + (rand() % (timers[timerid].icur - min));
+
+  // periodic timer
+  if (imin == imax) {
+    xTimerChangePeriod(timers[timerid].timer, pdMS_TO_TICKS(imin * 1000), 0);
```

Comment:
That's not how trickle timers are working.

I varies between imin and imax. The timer period is then randomly picked between I/2 and I. If imin = imax, the timer expires after a random period picket between I/2 and I, I being always equal to imin and imax.

Reply:
> That's not how trickle timers are working.
> 
> I varies between imin and imax. The timer period is then randomly picked between I/2 and I. If imin = imax, the timer expires after a random period picket between I/2 and I, I being always equal to imin and imax.
> 
> The period always has some randomness.

Sure, because it's for metrics sendings. I implemented it for metrics sendings, since it should be periodical constant time in FND. Registration timer is a trickle timer. Without this modification, we cannot send the metrics in every 15 minutes (default and the lowest settings in FND) constantly, original Linux app neither. I didn't modify linux and freertos on linux osal implementation accordingly.

Reply:
Workaround to make metrics timer periodic has been removed